lib/preview/shadowDevServer.js

```diff
@@ -0,0 +1,86 @@
+const net = require('net');
+const express = require('express');
+const { Router } = require('express');
+const cors = require("cors");
+
+const SHADOW_PORT = 1442;
+
+const startShadowDevServer = async (sessionInfo) => {
+  const portIsTaken = await new Promise((res, rej) => {
+    console.log(`Starting proxy link server on port ${SHADOW_PORT}`);
+    const testNetServer = net.createServer();
+
+    testNetServer.once('error', err => {
+      if (err['code'] === 'EADDRINUSE') {
+        console.error(
+          `Port ${SHADOW_PORT} is in use. HubSpot is unable to automatically create proxy links in the Sprocket Menu`
+        );
+        res(true);
+      } else {
+        rej(err);
+      }
+    });
+
+    testNetServer.once('listening', () => {
+      testNetServer.close();
+    });
+
+    testNetServer.once('close', () => {
+      res(false);
+    });
+
+    testNetServer.listen(SHADOW_PORT);
+  });
+
+  if (portIsTaken) {
+    return;
+  }
+
+  const shadowDevServer = express();
```

Comment:
Just curious, what does "shadow" here mean?

Reply:
I just borrowed this term from the JSR local dev server - definitely not the most explanatory variable name. Basically since the user can choose their own port for their dev server, we want a static port to listen on as well so that the sprocket menu on live sites have a place to ping on render to see if its running, and if so add a link to it that page in the local dev server.

Changing the name of that and associated vars to `sprocketMenuServer` to be a little clearer what its related to